### DISCUSSION TITLE
How to debug?

### DISCUSSION THREAD

**Discussion by 18601673727** [DioxusLabs/dioxus#136] [Q&A] [answered]
Very new to WASM or dioxus, after read the docs a little bit, run `trunk serve` with following code, nothing get printed out.
```rs
input {
    name: "test",
    r#type: "text",
    value: "",
    oninput: move || {
        println!("123"); // not printing anything
    },
}
```
Edit: forgot to mention that I'm running this on `web`.

**Comment by jkelleyrtp** [comment, marked as answer]
Check out 

- the log crate
- the wasmlogger crate
- console error panic hook crate

In wasm, you want to use the log crate instead of println.

**Reply by 18601673727** [reply]
Thanks for the kind help or the brilliant framework! Would you mind create a integrated repo for starters like me to easily jump start with all those basic stuff around? I tried log crate still not printing anything out..

**Reply by mrxiaozhuox** [reply]
> Thanks for the kind help or the brilliant framework! Would you mind create a integrated repo for starters like me to easily jump start with all those basic stuff around? I tried log crate still not printing anything out..

I think it's a great idea!

**Reply by t1m0t** [reply]
> Thanks for the kind help or the brilliant framework! Would you mind create a integrated repo for starters like me to easily jump start with all those basic stuff around? I tried log crate still not printing anything out..

You need to first initialize the wasm logger with this line in the `main()` function:
`wasm_logger::init(wasm_logger::Config::new(log::Level::Info));`
The level can be changed to `Trace` for very detailed ones.

You also need to understand that for the web, you basically rely on WASM so you don't want to print anything in stdout (except if it is what you really want).
